@here/harp-geometry/lib/OrientedBox3.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OrientedBox3Like } from "@here/harp-geoutils";
+import { Frustum, Matrix4, Plane, Vector3 } from "three";
+
+export class OrientedBox3 implements OrientedBox3Like {
+    /**
+     * The position of the center of this `OrientedBox3`.
+     */
+    readonly position = new Vector3();
+
+    /**
+     * The x-axis of this `OrientedBox3`.
+     */
+    readonly xAxis = new Vector3(1, 0, 0);
```

Comment:
> `xAxis`
> `yAxis`
> `zAxis`

Having threee independent vectors implies we support skewed boxes, so not only box (which is by intuition cuboid ...).
So do we support non-cuboid boxes then ?

Should this be optimized-out to only one rotation vector or is this too complex and not worth effort ?